gengokit/httptransport/httptransport.go

```diff
@@ -101,6 +111,24 @@ func NewBinding(i int, meth *deftree.ServiceMethod) *Binding {
 		nField.LowCamelName = LowCamelName(nField.Name)
 
 		nBinding.Fields = append(nBinding.Fields, &nField)
+
+		// Emit warnings for certain cases
+		if nField.IsBaseType == false {
+			wrn := []string{
+				"%s.%s is a custom type '%s', only base types and repeated base ",
+				"types are supported. As a result, the generated HTTP ",
+				"transport will fail to compile. Remove non-base types.",
+			}
+			log.Warnf(strings.Join(wrn, ""), meth.GetName(), nField.Name, nField.ProtobufType)
+		}
+		if field.Label == "LABEL_REPEATED" && nField.Location == "path" {
+			wrn := []string{
+				"%s.%s is a repeated field specified to be in the path. ",
+				"Currently, repeated fields are not supported in the path ",
```

Comment:
Wait, won't maps and nested messages work if they are json in the body of a http request? It will just be unmarshaled.